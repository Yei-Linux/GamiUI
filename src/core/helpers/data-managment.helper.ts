export const mapEnum = (enumerable: any): any[] => {
  const enumMembers: any[] = Object.keys(enumerable).map(
    (key) => enumerable[key]
  )
  return enumMembers
}
