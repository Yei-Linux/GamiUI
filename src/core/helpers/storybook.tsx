import React from "react";
import TableStories from "../../styles/utilities/TableStories";
import { options as optionsStyle } from "../utils/constants";

export const getTemplate = (Component: React.ComponentType<any>): any => (
  args: any
) => {
  return <Component {...args} />;
};

export const getListTemplate = (Component: React.ComponentType<any>): any => ({
  items,
  field,
  ...args
}: {
  items: any[];
  field: any;
}) =>
  items.map((item, index) => (
    <TableStories key={index} item={item} field={field}>
      <Component {...args} {...item} />
    </TableStories>
  ));

export const stylesControl = {
  style: { control: "object" },
  textAlign: { control: { type: "select", options: optionsStyle.textAlign } },
  fontWeight: {
    control: { type: "select", options: optionsStyle.fontWeight },
  },
  width: { control: { type: "select", options: optionsStyle.width } },
  shadow: { control: { type: "select", options: optionsStyle.shadow } },
  border: { control: { type: "select", options: optionsStyle.border } },
  heigth: { control: { type: "select", options: optionsStyle.height } },
};