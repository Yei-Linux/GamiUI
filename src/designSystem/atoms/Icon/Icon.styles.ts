import styled from "@emotion/styled";
import { twinStyles } from "../../../styles/utilities/twinStyles";

export const IconWrapper = styled.i<{
  border?: string;
  shadow?: string;
}>`
  border: none;
  outline: none;
  cursor: pointer;
  appearance: none;
  display: flex;
  justify-content: center;
  align-items: center;

  ${(props: any) => twinStyles(props)};
`;
