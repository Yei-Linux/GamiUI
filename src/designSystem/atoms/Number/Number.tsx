import React from "react";
import Input from "../Input";
import { IInput } from "../Input/Input";

const REGEX__NUMBER = /^[0-9]+$/;

export interface INumber extends IInput {}

const Number = ({ onChangeFormItem, ...args }: INumber) => {
  const handleTypying = (value: any) => {
    if (REGEX__NUMBER.test(value)) {
      onChangeFormItem(value);
    }
  };

  return <Input onChangeFormItem={handleTypying} {...args} />;
};

Number.defaultProps = {
  width: "NORMAL",
  heigth: "SMALL",
  border: "MEDIUM",
};

export default Number;
