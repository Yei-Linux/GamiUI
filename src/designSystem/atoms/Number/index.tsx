export { default } from './Number'
export { options } from './constants'