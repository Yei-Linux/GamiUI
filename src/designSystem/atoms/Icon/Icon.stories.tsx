import { Meta } from "@storybook/react";
import { getTemplate, stylesControl } from "../../../core/helpers/storybook";

import Icon from ".";
import { options as optionsIcon } from "./constants";

const Template = getTemplate(Icon);

export default {
  title: "Atoms/Icon",
  component: Icon,
  argTypes: {
    children: { control: "text" },
    name: { control: { type: "select", options: optionsIcon.names } },
    ...stylesControl,
  },
} as Meta;

export const BasicIcon = Template.bind({});
BasicIcon.args = {
  fill: "red",
  size: "25px",
  name: "facebook",
};
