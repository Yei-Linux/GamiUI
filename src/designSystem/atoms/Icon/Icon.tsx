import React, { useEffect, useState } from "react";
import { IGeneralProps } from "../../../core/domain/interfaces/IGeneralProps";
import { IconNames } from "../../../core/domain/types";
import { IconsPack } from "./constants";
import { IconWrapper } from "./Icon.styles";

interface IIcon extends IGeneralProps {
  /**
   * Fill Icon
   */
  fill: string;
  /**
   * Name
   */
  name: IconNames;
  /**
   * Size icon
   */
  size: string;
}

const Svg = ({
  children,
  ...props
}: {
  children: React.ReactNode;
  width: string;
  height: string;
  viewBox: string;
  fill: string;
}) => {
  return (
    <svg xmlns="http://www.w3.org/2000/svg" {...props}>
      {children}
    </svg>
  );
};

const Icon = ({ fill, name, size, ...args }: IIcon) => {
  const [icon, setIcon] = useState(IconsPack[name]);

  useEffect(() => {
    setIcon(IconsPack[name]);
  }, [name]);

  return (
    <IconWrapper {...args}>
      <Svg fill={fill} width={size} height={size} viewBox={icon.viewBox}>
        {icon.svg}
      </Svg>
    </IconWrapper>
  );
};

Icon.defaultProps = {
  fill: "none",
  name: "facebook",
  size: "16px",
};

export default Icon;