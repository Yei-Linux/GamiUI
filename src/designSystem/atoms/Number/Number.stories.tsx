import { Meta } from "@storybook/react";
import { getTemplate, stylesControl } from "../../../core/helpers/storybook";

import Number from ".";
import React from "react";
import Icon from "../Icon";

const Template = getTemplate(Number);

export default {
  title: "Atoms/Number",
  component: Number,
  args: { ...Number.defaultProps },
  argTypes: {
    positionPrefix: {
      control: "select",
      options: ["left", "right"],
    },
    placeholder: { control: "text" },
    prefix: {
      options: ["a", "b"],
      mapping: {
        a: null,
        b: <Icon fill="#374a54" />,
      },
      control: {
        type: "select",
        labels: {
          a: "withoutIcon",
          b: "withIcon",
        },
      },
    },
    ...stylesControl,
  },
} as Meta;

export const Basic = Template.bind({});
Basic.args = {
  placeholder: "Type your phone",
  width: "NORMAL",
  heigth: "SMALL",
  shadow: "",
  border: "MEDIUM",
  prefix: <Icon fill="#374a54" />,
};
