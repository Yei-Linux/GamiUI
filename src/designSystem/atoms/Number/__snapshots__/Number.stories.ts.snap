// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Storyshots Atoms/Number Basic 1`] = `
<div
  autoComplete="off"
  className="css-1hwl8w3-InputWrapper e7vlt9n0"
  name="gamification"
  placeholder="Type your phone"
  prefix={
    <Icon
      fill="#374a54"
      name="facebook"
      size="16px"
    />
  }
  readOnly={false}
  type="input"
  width="NORMAL"
>
  <i
    className="css-6tw2qr-IconWrapper e1cr9qb30"
  >
    <svg
      fill="#374a54"
      height="16px"
      viewBox="0 0 20 20"
      width="16px"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M8.46 18H11.39V10.7H13.84L14.21 7.86H11.39V6.04C11.39 5.22 11.62 4.66 12.8 4.66H14.31V2.11C14.05 2.08 13.16 2 12.12 2C9.94 2 8.46 3.33 8.46 5.76V7.86H6V10.7H8.46V18Z"
      />
    </svg>
  </i>
  <input
    autoComplete="off"
    className="css-7dwcir-InputContainer e7vlt9n1"
    name="gamification"
    onChange={[Function]}
    placeholder="Type your phone"
    prefix={
      <Icon
        fill="#374a54"
        name="facebook"
        size="16px"
      />
    }
    readOnly={false}
    type="input"
    width="NORMAL"
  />
</div>
`;
